### PR TITLE
base_tests.sh: make the test self-contained

### PR DESCRIPTION
**Pull request by msehnout**
We claim to have self-contained test cases, but the base_tests.sh script
still requires the WORKSPACE environment variable to be set outside of
the script, which is what Jenkins does.

This patch introduces a default value, which makes it possible to use
the script outside of Jenkins.

### CODE REVIEW

**Review by ondrejbudai — APPROVED** [review]
Reasonable, thanks!

**Comment by msehnout** [comment]
It failed on the auth-test
```
[2020-11-04T09:41:04.292Z] === RUN   TestWorkerAPIAuth/certificate_signed_by_a_trusted_CA/valid_CN_1
[2020-11-04T09:41:04.292Z] Signature ok
[2020-11-04T09:41:04.292Z] subject=CN = worker.osbuild.org
[2020-11-04T09:41:04.292Z] Getting CA Private Key
[2020-11-04T09:41:04.293Z]     TestWorkerAPIAuth/certificate_signed_by_a_trusted_CA/valid_CN_1: main_test.go:240: 
[2020-11-04T09:41:04.293Z]         	Error Trace:	main_test.go:240
[2020-11-04T09:41:04.293Z]         	            				main_test.go:150
[2020-11-04T09:41:04.293Z]         	Error:      	Received unexpected error:
[2020-11-04T09:41:04.293Z]         	            	Get "https://localhost:8700/api/worker/v1/status": x509: certificate has expired or is not yet valid: current time 2020-11-04T09:41:04Z is after 2020-11-04T08:38:53Z
[2020-11-04T09:41:04.293Z]         	Test:       	TestWorkerAPIAuth/certificate_signed_by_a_trusted_CA/valid_CN_1
[2020-11-04T09:41:04.293Z] === RUN   TestWorkerAPIAuth/certificate_signed_by_a_trusted_CA/valid_CN_2
[2020-11-04T09:41:04.698Z] Signature ok
[2020-11-04T09:41:04.698Z] subject=CN = localhost
[2020-11-04T09:41:04.698Z] Getting CA Private Key
[2020-11-04T09:41:04.698Z]     TestWorkerAPIAuth/certificate_signed_by_a_trusted_CA/valid_CN_2: main_test.go:240: 
[2020-11-04T09:41:04.698Z]         	Error Trace:	main_test.go:240
[2020-11-04T09:41:04.698Z]         	            				main_test.go:150
[2020-11-04T09:41:04.698Z]         	Error:      	Received unexpected error:
[2020-11-04T09:41:04.698Z]         	            	Get "https://localhost:8700/api/worker/v1/status": x509: certificate has expired or is not yet valid: current time 2020-11-04T09:41:04Z is after 2020-11-04T08:38:53Z
[2020-11-04T09:41:04.698Z]         	Test:       	TestWorkerAPIAuth/certificate_signed_by_a_trusted_CA/valid_CN_2
[2020-11-04T09:41:04.698Z] === RUN   TestWorkerAPIAuth/certificate_signed_by_a_trusted_CA/invalid_CN
```
thoughts, @ondrejbudai ?

**Comment by ondrejbudai** [comment]
`master` is green, this PR is not, therefore you must have broken it. :P

Just kidding, the certs expired, #1057 solves this, it's just waiting for Lars' review.

File: test/cases/base_tests.sh
Comment (teg) [inline test/cases/base_tests.sh:30]: Two suggestions:
 - drop $WORKSPACE entirely (also in other places, if it is used, but that could be independent :) )
 - use a proper temporary directory that is created with a random name and cleaned up, otherwise we risk tests writing stuff on top of each other

**Review by teg — CHANGES_REQUESTED** [review]
Thanks for this! Just minor suggestions.

**Comment by ondrejbudai** [comment]
This needs a rebase after #1063 lands.

File: test/cases/base_tests.sh
Comment (msehnout) [reply test/cases/base_tests.sh:30]: Makes sense, but this will make logs collection more complicated. See the updated patch.

**Review by ondrejbudai — COMMENTED** [review]
Looks good to me, @teg?

**Review by teg — APPROVED** [review]
Thanks!

**Pull request merged**
